internal/prober/multihttp/script.go

```diff
@@ -17,13 +18,37 @@ import (
 //go:embed script.tmpl
 var templateFS embed.FS
 
-func buildUrl(req *sm.MultiHttpEntryRequest) string {
-	// If we are here, the request has already been validated, and the URL
-	// should be valid. This function should never return an error.
-	u, _ := url.Parse(req.Url)
+func replaceVariablesInString(s string) string {
+	re := regexp.MustCompile(`\$\{(.*?)\}`)
```

Comment:
the capture probably needs to be something like `[a-zA-Z_][a-zA-Z0-9_]*`

Reply:
given some input like `foo${bar}baz${bar}` this will return `bar` twice.

This should take care of that:

```
slices.Sort(names)
names = slices.Compact(names)
```